cypress-io/cypress
# fix(cypress-schematic): suffix template files so they are not ignored

### User facing changelog
Fix cypress schematic files that were not being included in the published npm package

__NOTE__: Not applicable to Cypress changelog

### Additional details
The folder structure for the schematic template files was recently updated but the `.npmignore` pattern was not, causing all of the `.ts` files to not be included in the published package. I opted to rename all of the `.ts` files to use the `.template` suffix, since they can contain invalid syntax and shouldn't be typechecked

### Steps to test

With latest published package:

1. Create a new angular cli application with `ng new my-app` (might need to `npm i -g @angular/cli`)
2. `cd my-app` and run `ng add @cypress/schematic`
3. Follow prompts (no CT testing), and check the `cypress` folder, it will be missing `e2e/spec.cy.ts` and `support/**`
4. Look in `node_modules/@cypress/schematic/src/schematics/ng-add/files-core`, it is missing the relevant files

With local (my branch)

1. In Cypress repo: `cd npm/cypress-schematic && npm pack`
2. Inside new Angular app, run `npm i @cypress/schematic@file:<absolute-path-to-tgz>cypress-schematic-0.0.0-development.tgz`
3. Revert changes made in angular app and run `ng add @cypress/schematic`
4. Verify files are scaffolded correctly, check in node_modules for properly included files

### How has the user experience changed?
Before:
![Screen Shot 2022-08-31 at 9 45 46 AM](https://user-images.githubusercontent.com/25158820/187707719-16181b6a-7c99-4051-9e01-2fc5ef0de6f1.png)

After:
![Screen Shot 2022-08-31 at 9 46 34 AM](https://user-images.githubusercontent.com/25158820/187707914-c3278e6c-8b9d-4a43-ac00-d0b383546055.png)


### PR Tasks


- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
